master/pkg/model/experiment_config.go

```diff
@@ -274,3 +275,58 @@ type InternalConfig struct {
 type NativeConfig struct {
 	Command []string `json:"command"`
 }
+
+// DebugConfig represents which subsystems will be debugged in the harness.
+type DebugConfig struct {
+	RootLogLevel             *string `json:"root_log_level"`
+	StorageLogLevel          *string `json:"storage_log_level"`
+	HarnessLogLevel          *string `json:"harness_log_level"`
+	DebugAllWorkers          bool    `json:"debug_all_workers"`
+	HorovodVerbose           bool    `json:"horovod_verbose"`
+	NCCLDebug                *string `json:"nccl_debug"`
+	NCCLDebugSubsys          *string `json:"nccl_debug_subsys"`
+	ResourceProfilePeriodSec float32 `json:"resource_profile_period_sec"`
+	StackTracePeriodSec      float32 `json:"stack_trace_period_sec"`
+}
+
+var (
+	debugConfigInfo  = "INFO"
+	debugConfigDebug = "DEBUG"
+	// trueDebugConfig is the value of DebugConfig that we assign for `debug: true`.
+	trueDebugConfig = DebugConfig{
+		RootLogLevel:             &debugConfigDebug,
+		StorageLogLevel:          &debugConfigInfo,
+		HarnessLogLevel:          &debugConfigDebug,
+		DebugAllWorkers:          true,
+		HorovodVerbose:           true,
+		NCCLDebug:                nil,
```

Comment:
(see previous comment on the default behavior on the python side, but I'm not going to resolve this comment so that after we resolve that I remember to fix the default behavior on both sides)